packages/compiler-cli/test/ngtsc/ngtsc_spec.ts

```diff
@@ -5222,6 +5222,244 @@ export const Foo = Foo__PRE_R3__;
       });
     });
 
+    describe('undecorated providers', () => {
+      it('should error when an undecorated class is provided directly in a module', () => {
```

Comment:
I feel that we need to be more clear here, since a class that has no constructor or a constructor with no parameters IS allowed.

```suggestion
      it('should error when an undecorated class, with a non-trivial constructor, is provided directly in a module', () => {
```